src/app/shared/navigation-focus/navigation-focus.service.ts

```diff
@@ -0,0 +1,67 @@
+import {ElementRef, Injectable, OnDestroy} from '@angular/core';
+import {Event, NavigationEnd, Router} from '@angular/router';
+import {filter, skip} from 'rxjs/operators';
+import {Subscription} from 'rxjs';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class NavigationFocusService implements OnDestroy {
+  private subscriptions = new Subscription();
+  private navigationFocusRequests: ElementRef[] = [];
+  private skipLinkFocusRequests: ElementRef[] = [];
+  private skipLinkHref = '';
+
+  readonly navigationEndEvents = this.router.events
+    .pipe(filter((event: Event): event is NavigationEnd => event instanceof NavigationEnd));
+  readonly softNavigations = this.navigationEndEvents.pipe(skip(1));
+
+  constructor(private router: Router) {
+    this.subscriptions.add(this.softNavigations.subscribe(() => {
+      // focus if url does not have fragment
+      if (!this.router.url.split('#')[1]) {
+        setTimeout(() => {
+          if (this.navigationFocusRequests.length) {
+            this.navigationFocusRequests[this.navigationFocusRequests.length - 1]
+              .nativeElement.focus({preventScroll: true});
+          }
+        }, 100);
+      }
+    }));
+  }
+
+  ngOnDestroy() {
+    this.subscriptions.unsubscribe();
+  }
+
+  requestFocusOnNavigation(el: ElementRef, wantsFocus: boolean) {
+    wantsFocus ? this.navigationFocusRequests.push(el) :
+      this.navigationFocusRequests.splice(this.navigationFocusRequests.indexOf(el), 1);
+  }
+
+  requestSkipLinkFocus(el: ElementRef, wantsFocus: boolean) {
```

Comment:
Can we refactor these two be different functions (2 instead of 4) without the boolean? The function bodies have completely different behaviors based on the boolean value and don't share any code. I feel like this is the purest example of the boolean argument as a "code smell".